eth_account/messages.py

```diff
@@ -14,14 +14,23 @@
 )
 
 
-def defunct_hash_message(primitive=None, hexstr=None, text=None):
+def defunct_hash_message(
+        *,
+        primitive=None,
+        hexstr=None,
+        text=None,
+        signature_version=b'E',
+        version_specific_data=None):
     '''
     Convert the provided message into a message hash, to be signed.
     This provides the same prefix and hashing approach as
-    :meth:`w3.eth.sign() <web3.eth.Eth.sign>`. That means that the
-    message will automatically be prepended with text
-    defined in EIP-191 as version 'E': ``b'\\x19Ethereum Signed Message:\\n'``
+    :meth:`w3.eth.sign() <web3.eth.Eth.sign>`. The message will by default
+    be prepended with text defined in EIP-191 as
+    version 'E': ``b'\\x19Ethereum Signed Message:\\n'``
     concatenated with the number of bytes in the message.
+    Currently you can only specify the ``signature_version`` as
+    version ``0``: Sign data with intended validator (EIP 191).
```

Comment:
`signature_version` here can be either version `0` or version `E`, right? We should probably update this sentence to reflect that. 

Maybe something like:
Currently, the only two supported signature versions are `E` and `0`. If version `0` is used, the `version_specific_data` is the address of the intended validator.

Reply:
I'd avoid referring to it as `E`, and rather to represent both as `0x00` and `0x45` to make it clear that it's their *byte* value that matters.  After representing them that way, you could make the connection that `0x45` is equivalent to the ASCII letter `E`